pkg/aws/policies.go

```diff
@@ -634,69 +634,75 @@ func isAccountRoleVersionCompatible(tagsList []*iam.Tag, roleType string,
 	return true, nil
 }
 
-func (c *awsClient) ListRoles() ([]*iam.Role, error) {
-	roles := []*iam.Role{}
-	err := c.iamClient.ListRolesPages(&iam.ListRolesInput{}, func(page *iam.ListRolesOutput, lastPage bool) bool {
-		roles = append(roles, page.Roles...)
-		return aws.BoolValue(page.IsTruncated)
-	})
-	return roles, err
+func (c *awsClient) ListRoles() ([]*iamtypes.Role, error) {
+	roles := []*iamtypes.Role{}
+	paginator := iam.NewListRolesPaginator(c.iamClient, &iam.ListRolesInput{})
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(context.TODO())
+		if err != nil {
+			return nil, err
+		}
+		for _, role := range output.Roles {
+			r := role
+			roles = append(roles, &r)
```

Comment:
Should we return a slice of roles like the AWS SDK is now returning? instead of pointers